crates/locutus-node/src/operations/subscribe.rs

```diff
@@ -426,8 +489,108 @@ mod messages {
 
 #[cfg(test)]
 mod test {
+    use super::*;
+    use crate::contract::Contract;
+    use crate::ring::Location;
+    use crate::{conn_manager::PeerKey, node::SimStorageError};
+
```

Comment:
would be nice if we add an other test where we emulate the failure to get contract at the first node and forwarding to a third node.